lib/api/2.0/notification.js

```diff
@@ -30,7 +36,124 @@ var notificationProgressPost = controller(function(req, res){
+         "Severity": "Critical",
+         "sourceIpAddress": "10.240.19.130",
+         "rackhdId": "58d94cec316779d4126be134",
+         "ChassisName": "PowerEdge R630",
+         "ServiceTag": "4666482",
+         "SN": "CN747515A80855"
+         },
+     "severity": "critical",
+     "version": "1.0",
+     "createdAt": "2017-03-29T19:44:12.972Z"
+     }
+     */
+
+    return Promise.try(function(){
+        var ip = req.headers['x-forwarded-for'] ||
+            req.connection.remoteAddress ||
+            req.socket.remoteAddress ||
+            req.connection.socket.remoteAddress;
+        var amqpMessage = {};
+        var deviceInfoCatalog ={};
+        var bmcCatalog = {};
+        var alert = {};
+
+        return new Promise(function(resolve, reject) {
+            var parse = parser.urlencoded({ extended: false });
+            parse(req, res, function (err) {
+                if (err) {
+                    return reject(err);
+                }
+                    resolve();
+            });
+        })
+            .then(function() {
+            return waterline.catalogs.findOne({source: "bmc", "data.IP Address": ip})
+                .then(function(data) {
+                    if (data !== undefined && data.hasOwnProperty("node") && data.node !== ""){
+                         bmcCatalog = data;
+                        //Format the amqp message by adopting the rackHD notification standard
+                        amqpMessage.type = "node";
+                        amqpMessage.action = "alerts";
+                        amqpMessage.typeId = bmcCatalog.node;
+
+                        var httpReq = _.defaults(req.swagger.query || {},
+                            req.query || {},
+                            req.body || {});
+                        if(Object.keys(httpReq).length === 1 ) {
+                            var key = Object.keys(httpReq)[0];
+                            alert = JSON.parse(key);
+                            amqpMessage.data = alert;
+                            amqpMessage.data.sourceIpAddress = ip;
+                            return waterline.catalogs.findOne({source: "ipmi-fru"})
+                                .then(function(catalogIpmiFru){
+                                    if(catalogIpmiFru !== undefined &&
+                                        catalogIpmiFru.data
+                                            .hasOwnProperty("Builtin FRU Device (ID 0)")){
+                                        deviceInfoCatalog =
+                                            catalogIpmiFru.data["Builtin FRU Device (ID 0)"];
+
+                                        //Add useful information to the alert message
+                                        amqpMessage.data.rackhdId = bmcCatalog.node;
+                                        amqpMessage.data.macAddress = bmcCatalog["MAC Address"];
+                                        amqpMessage.severity = alert.Severity.toLowerCase();
```

Comment:
Should above 3 sentences moved out of this if block? Those assignments should still be valid even if there is no ipmi fru catalog.